chartfx-chart/src/test/java/de/gsi/chart/ui/css/CssPropertyFactoryTest.java

```diff
@@ -0,0 +1,161 @@
+
+        // test applying via css sheet
+        // test with multiple instances
+    }
+
+    public static class TestStyleable extends Region {
+        private static final CssPropertyFactory<TestStyleable> CSS = new CssPropertyFactory<>(Region.getClassCssMetaData());
+
+        @Override
+        public List<CssMetaData<? extends Styleable, ?>> getCssMetaData() {
+            return CSS.getCssMetaData();
+        }
+
+        // double
+        private final StyleableDoubleProperty testDouble = CSS.createDoubleProperty(this, "testDouble", "-fx-test-double", s -> s.testDouble, 4.2, true, null);
+
+        public double getTestDouble() {
+            return testDouble.getValue();
+        }
+
+        // float
+        private final StyleableFloatProperty testFloat = CSS.createFloatProperty(this, "testFloat", "-fx-test-float", s -> s.testFloat, 1.337f, true, null);
+
+        public float getTestFloat() {
+            return testFloat.getValue();
+        }
+
+        // int
+        private final StyleableIntegerProperty testInt = CSS.createIntegerProperty(this, "testInt", "-fx-test-int", s -> s.testInt, 42, true, null);
+
+        public int getTestInt() {
+            return testInt.getValue();
+        }
+
+        // long
+        private final StyleableLongProperty testLong = CSS.createLongProperty(this, "testLong", "-fx-test-long", s -> s.testLong, 123456789123456789l, true, null);
+
+        public long getTestLong() {
+            return testLong.getValue();
+        }
+
+        // string
+        private final StyleableStringProperty testString = CSS.createStringProperty(this, "testString", "-fx-test-string", s -> s.testString, "test", true, null);
+
+        public String getTestString() {
+            return testString.getValue();
+        }
+
+        // boolean
+        private final StyleableBooleanProperty testBool = CSS.createBooleanProperty(this, "testBool", "-fx-test-bool", s -> s.testBool, false, true, null);
+
+        public boolean getTestBool() {
+            return testBool.getValue();
+        }
+
+        // object
+        private final StyleableObjectProperty<Paint> testObject = CSS.createObjectProperty(this, "testObject", "-fx-test-object", s -> s.testObject, Color.PINK, true, StyleConverter.getPaintConverter(), null);
+
+        public Paint getTestObject() {
+            return testObject.getValue();
+        }
+
+        // enum
+        private final StyleableObjectProperty<TestEnum> testEnum = CSS.createObjectProperty(this, "testEnum", "-fx-test-enum", s -> s.testEnum, TestEnum.ENABLED, true, StyleConverter.getEnumConverter(TestEnum.class), null);
+
+        public TestEnum getTestEnum() {
+            return testEnum.getValue();
+        }
+
+        // enum with pseudoclass
+        private final StyleableObjectProperty<TestEnum> testEnumP = CSS.createEnumPropertyWithPseudoclasses(this, "testEnumP", "-fx-test-enum-p",
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5741094)